src/main/java/cceclipseplugin/core/EclipseRequestManager.java

```diff
@@ -160,7 +160,7 @@ public void pullFileAndCreate(IProject p, Project ccp, File file, IProgressMonit
 						meta.setFilename(file.getFilename());
 						meta.setRelativePath(file.getRelativePath());
 						meta.setVersion(file.getFileVersion());
-						PluginManager.getInstance().getMetadataManager().putFileMetadata(newFile.getFullPath().removeLastSegments(1).toString(), 
+						PluginManager.getInstance().getMetadataManager().putFileMetadata(newFile.getLocation().toString(), 
```

Comment:
There are several places in this function where you use `PluginManager.getInstance()`. Can you just make a local variable for it in the beginning of the function?